File: java/src/test/java/problems1_100/SameTreeTest.java
package problems1_100;

import concepts.TreeNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class SameTreeTest extends SampledTest {
    @Test
    public void test() {
        for (SameTree solution : SameTree.values()) {
            for (SameTreeSample sample : getSamples(SameTreeSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.isSameTree(
                            TreeNode.of(sample.input.p),
                            TreeNode.of(sample.input.q)
                        )
                    ).isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/SameTreeTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class SameTreeTest : SampledTest() {
    @Test
    fun test() =
        SameTree.entries.forEach { solution ->
            getSamples<Array<SameTreeSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(
                        solution.isSameTree(
                            TreeNode.of(*input.p),
                            TreeNode.of(*input.q),
                        ),
                    ).isEqualTo(output)
            }
        }
}
